src/fragment.js

```diff
@@ -42,6 +44,11 @@ const absolute = (ComposedComponent: ReactClass<*>) => {
 
 const relative = (ComposedComponent: ReactClass<*>) => {
   class RelativeFragment extends Component {
+    constructor() {
+      super();
+      this.id = generateId();
```

Comment:
With React's client/server reconciliation? I don't think so, since it's not a part of the render output at all, it's just an internal id that determines if it should render in the first place.